ui/src/pages/backoffice/Item/ItemForms/ItemEditor/components/ItemForm/ItemForm.js

```diff
@@ -24,12 +25,52 @@ import {
   serializeDocument,
   serializeInternalLocation,
 } from '@components/ESSelector/serializer';
+import { vocabulary as vocabularyApi } from '@api';
 
 export class ItemForm extends Component {
   config = invenioConfig.items;
+  state = {
+    currencies: [],
+  };
+
+  componentDidMount() {
+    this.fetchCurrencies();
+  }
+
+  query = () => {
+    const searchQuery = vocabularyApi
+      .query()
+      .withType(invenioConfig.vocabularies.currencies)
+      .qs();
+    return vocabularyApi.list(searchQuery);
+  };
+
+  serializer = hit => ({
+    key: hit.metadata.key,
+    value: hit.metadata.key,
+    text: hit.metadata.key,
+  });
+
+  fetchCurrencies = async () => {
+    try {
+      const response = await this.query();
+      const currencies = response.data.hits.map(hit => this.serializer(hit));
+      this.setState({ isLoading: false, currencies: currencies, error: null });
```

Comment:
Major: you have probably found this somewhere else :), and I am sorry about that, but this can unfortunately introduce memory leaks.
I would like to explain you why and then we will need to change this. 2 options:
1. use Redux instead
2. use [withCancel](https://github.com/inveniosoftware/invenio-app-ils/blob/master/ui/src/api/utils.js#L122)

We will also have to change any occurrence of this wrong code.